src/uhttp.c

```diff
@@ -320,7 +325,48 @@ static int write_text_line(HTTP_CLIENT_HANDLE_DATA* http_data, const char* text_
         result = 0;
         if (http_data->trace_on)
         {
-            LOG(AZ_LOG_TRACE, LOG_LINE, "%s", text_line);
+            char* authStart = strstr(text_line, AUTH_HEADER);
+            char* proxyAuthStart = strstr(text_line, PROXY_AUTH_HEADER);
+            char* authEol = strstr(authStart, HEADER_ENDING);
+            char* proxyAuthEol = strstr(proxyAuthStart, HEADER_ENDING);
+            if (authStart == NULL && proxyAuthStart == NULL)
```

Comment:
Can you add unit tests for these 5 scenarios since there's a lot of string manipulation going on? (both null, only proxy null, only auth null, neither null with proxy first, and neither null with proxy second)

Reply:
I won't be adding unit test since this is related to logging (per Ewerton's advice). But I am doing individual test for each scenario with a sample.

Reply:
Just chiming in here, this function (log_text_line) is static, so it is not directly testable through unit tests. We can add tests calling the public functions that indirectly call log_text_line(), but we also do not unit-test logging in the SDK. So these unit tests would not assert anything related to the new logging, but can be used for a manual verification in the console that the expected logs are being printed.
Please share any ideas if you think otherwise.